client/client.js

```diff
@@ -150,7 +150,8 @@ Client.prototype.serviceList = serviceList;
 function serviceDestroy(nameOrId, callback) {
   var Service = this.models.ServerService;
 
-  var q = {where: {or: [{name: nameOrId}, {id: nameOrId}]}};
+  var q = (typeof nameOrId === 'string') ? {where: {name: nameOrId}} :
+      {where: {or: [{name: nameOrId}, {id: nameOrId}]}};
```

Comment:
Should update all the other places in client.js that use the same filter style. Eg: service, group create/find/delete locations.